### ISSUE TITLE
[BUG] Out-of-bounds Fn::Select on literal list silently passes linting, causing deploy-time failure

### ISSUE THREAD

**Issue by mchittineni** [aws-cloudformation/cfn-lint#4693]
### CloudFormation Lint Version
v1.56.2

### What operating system are you using?
macOS / Linux / Windows

### Describe the bug
When `Fn::Select` is invoked with a literal index that is out of bounds for a literal list (e.g. `!Select [5, ["subnet-1", "subnet-2"]]`), `cfn-lint` reports **0 errors** and silently passes.

When deployed to CloudFormation, `CreateStack`/`UpdateStack` immediately fails and rolls back with:
```text
Template error: Fn::Select cannot select nonexistent value at index 5
```

#### Root Cause
In `src/cfnlint/jsonschema/_resolvers_cfn.py` lines 371-375:
https://github.com/aws-cloudformation/cfn-lint/blob/a4ed5bb8bba23101ff2a3b04c8ba5a3a290d20ef/src/cfnlint/jsonschema/_resolvers_cfn.py#L371-L375

```python
def select(validator: Validator, instance: Any) -> ResolutionResult:
    ...
    for i, _, _ in indexes:
        for obj, obj_v, _ in objs:
            try:
                i = int(i)
            except ValueError:
                continue
            if not validator.is_type(obj, "array"):
                continue
            if i < 0:
                continue
            if len(obj) <= i:
                continue
            yield from obj_v.resolve_value(obj[i])
```

When `len(obj) <= i`:
- `select()` simply executes `continue` and yields nothing.
- Neither `_resolvers_cfn.py` nor `src/cfnlint/rules/functions/Select.py` emits an error indicating that the index is out of bounds for the literal list.
- As a result, this fatal CloudFormation template error passes linting completely undetected.

### Expected behavior
`cfn-lint` should report a validation error when a literal index exceeds the length of a resolved literal array in `Fn::Select`.

### Reproduction template
```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MySubnetRouteTableAssociation:
    Type: AWS::EC2::SubnetRouteTableAssociation
    Properties:
      RouteTableId: rtb-12345678
      SubnetId: !Select
        - 5
        - - subnet-11111111
          - subnet-22222222
```

Command:
```bash
cfn-lint template.yaml
```

Actual output:
Exits `0` with no errors reported.

CloudFormation deploy result:
`CREATE_FAILED: Template error: Fn::Select cannot select nonexistent value at index 5`

### Proposed Fix
When `obj` is an array and `i >= len(obj)` in `select` (or in a child validator for `Select`), emit a `ValidationError`:
```python
if len(obj) <= i:
    yield (
        None,
        obj_v,
        ValidationError(
            f"Index {i} out of bounds for list of length {len(obj)}",
            path=deque([0]),
        ),
    )
    continue
```
